src/logic/safe/store/actions/transactions/utils/transactionHelpers.ts

```diff
@@ -48,8 +49,40 @@ export const isInnerTransaction = (tx: TxServiceModel | Transaction, safeAddress
   return isSameAddress && Number(tx.value) === 0
 }
 
-export const isCancelTransaction = (tx: TxServiceModel | Transaction, safeAddress: string): boolean => {
-  return isInnerTransaction(tx, safeAddress) && isEmptyData(tx.data)
+export const isCancelTransaction = (tx: TxServiceModel, safeAddress: string): boolean => {
+  if (!sameAddress(tx.to, safeAddress)) {
+    return false
+  }
+
+  if (Number(tx.value)) {
+    return false
+  }
+
+  if (tx.data && !isEmptyData(tx.data)) {
+    return false
+  }
+
+  if (tx.operation !== CALL) {
+    return false
+  }
+
+  if (tx.baseGas) {
+    return false
+  }
+
+  if (Number(tx.gasPrice)) {
+    return false
+  }
+
+  if (tx.gasToken && !isEmptyAddress(tx.gasToken)) {
```

Comment:
the existence check (`tx.gasToken &&`) is theoretically done also by `isEmptyAddress`, but doesn't harm to double check